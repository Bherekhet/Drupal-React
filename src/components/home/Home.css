#navColor{
    background: #543864;
}

/* -------------------------------------------------------------------- */
.intro {
    color: #543864;
    /* font-size: 1em; */
    margin-top: 2px;
    width: 100%;
    background: #dbd8e3;
    padding: 10px 30px;

}

.card {
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    transition: 0.3s;
    width: 30%;
    float: left;
    margin: 2px 16px;
    border-radius: 10px;
    flex: 0 1 24%;
}

img {
    display: block;
    border: 0;
    width: 100%;
}

img:hover {
    cursor: pointer;
}

.card-content {
	padding: 1.4em;
}

.card-content .title{
	font-size: 100%;
}
.card-content p {
	font-size: 80%;
}

.card:hover {
    box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
}

.contain {
    padding: 2px 16px;
    
}