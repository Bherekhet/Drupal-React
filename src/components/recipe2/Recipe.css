* {box-sizing:border-box}

.recipe {
    background: F4F6F6;
    max-width: 60%;
    margin: 0 auto;
    padding-top: 5%;
}

.rec-top {
    max-width: 50%;
    width: 40%;
    position: relative;
    margin: auto;
}
.buttons {
    position: relative;
    margin: 0 auto;
    max-width: 40%;
}
.prev, .next {
    cursor: pointer;
    position: absolute;
    width: auto;
    margin-top: -22px;
    padding: 16px;
    color: white;
    font-weight: bold;
    font-size: 18px;
    transition: 0.6s ease;
    border-radius: 0 3px 3px 0;
    user-select: none;
}

.next {
    right: 0;
    border-radius: 3px 0 0 3px;
}

.prev:hover, .next:hover {
    background-color: rgba(0,0,0,0.8);
}

.recipe-title {
    color: red;
    background: white;
    font-size: 15px;
    padding: 8px 10px;
    position: absolute;
    bottom: 6px;
    width: 100%;
    text-align: center;
}

.rec-detail {
    padding-top: 5%;
    margin: 0 auto;
}

.rec-ing {
    width: 60%;
    margin: 0 auto;
    border: 2px solid black;
    list-style-type: none;
    margin-bottom: 5%;
    padding: 2%;
}

.rec-ing p:nth-child(1) {
    text-align: center;    
}