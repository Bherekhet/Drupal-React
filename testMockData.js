//mock recipe and home page data for testing
 export const rec = [
    {
      body:"this is a body",
      field_images:"image section",
      field_ingredients:"1 2 3 4",
      field_summary:"summary section",
      title:"title section",
      node:"0"
    },
    {
      body:"this is a body",
      field_images:"image section",
      field_ingredients:"5 4 3 2",
      field_summary:"summary section",
      title:"title section",
      node:"2"
    }
]

export const hom = [
  {
    title: "title",
    body: "body",
    view_node: "node/1"
  }
]