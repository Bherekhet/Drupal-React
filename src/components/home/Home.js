import React from 'react';
import {Route, Switch, Link } from 'react-router-dom'
import Recipe from '../recipe/Recipe.js';
import './Home.css';

// this component load the home page of the react app
// it will include an intro section and gives a snapshot view of recipes fetched

class Home extends React.Component {
    constructor(props){
      super(props);
      this.state = {
        c: false,
        node: null
      };
  
      this.onRecipeClick = this.onRecipeClick.bind(this);
    }

    //function gets called when a recipe picture is clicked, updates 'recipeClicked' state 
    onRecipeClick(node){
      this.setState({
        recipeClicked: true,
        node: node
      });
    }
    
    render(){
      const {recipeClicked, node} = this.state;
      const {recipeData, homePage} = this.props;
      
      if (recipeClicked){
        return <Recipe recipeID={node} recipeData={recipeData}/> 
      }

      return( 
        <div className="main">
          {/* if homepage info is fetched, it gets rendered */}
          {homePage.length > 0 && 
            <p className="intro" dangerouslySetInnerHTML={{__html: homePage[0].body}}></p>
          }
          {/* if homepage info is not found, displays message below */}
          {homePage.length === 0 && 
          <p className="intro" >{"Oops no data found for homepage on content manager"}</p>}
           
          <div className="contain">
              {recipeData.length === 0 && 
                <p>{"Oops no data found for recipes on content manager"}</p>
              }
              { recipeData.map((item, index) => 
              <div  className="card">
                <div key={item.view_node} className="card-content">
                  <span><p className="title">{item.title}</p></span>
                  <span><p>{item.field_summary}</p></span>
                  {/* <a href={item.title} onClick={(e) => this.onRecipeClick(e, index)}> */}
                  <a href="#" onClick={() => this.onRecipeClick(index)}>
                    {/* only the first of the w2 images feteched will be displayed */}
                    <img src={process.env.REACT_APP_API_BASE+(item.field_images.slice(0, item.field_images.indexOf(',')))} alt={item.title}/>
                  </a>
                </div>
              </div>
              )}
          </div>
        </div>
      );
    }
}

export default Home;