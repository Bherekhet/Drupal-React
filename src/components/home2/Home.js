import React from 'react';
import Recipe from '../recipe/Recipe.js';
import './Home.css';


class Home extends React.Component {
    constructor(props){
      super(props);
  
      this.state = {
        recipeClicked: false,
        node: null
      };
  
      this.onRecipeClick = this.onRecipeClick.bind(this);
    }
  
    onRecipeClick(e, node){
      e.preventDefault();
      this.setState({
        recipeClicked: true,
        node: node
      });
    }
    
    render(){
      const {recipeClicked, node} = this.state;
      const recipeData = this.props.recipeData;
      
      if (recipeClicked){
        console.log(recipeData);
        return <Recipe recipeID={node} recipeData={recipeData}/>
      } 
      return( 
        <div className="main">
          <p className="intro">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
          <div className="contain">
              { recipeData.map((item, index) => 
              <div key={item.view_node} className="card">
                <div className="card-content">
                  <span><p className="title">{item.title}</p></span>
                  <span><p>{item.field_summary}</p></span>
                  <a href={item.title} onClick={(e) => this.onRecipeClick(e, index)}>
                    <img src={'http://gtest.dev.wwbtc.com'+item.field_images} alt={item.title}/>
                  </a>
                </div>
                
              </div>
              )}
          </div>
        </div>
      );
    }
}

export default Home;