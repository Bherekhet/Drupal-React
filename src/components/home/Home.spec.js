import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home.js';
import Recipe from '../recipe/Recipe'
import {mount} from 'enzyme';
import {hom, rec} from '../../../testMockData.js';

//snapshot test for Home component
describe('Home component', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Home homePage={hom} recipeData={rec}/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });


  // it('renders the inner Recipe component', () => {
  //   const wrapper = mount(<Home recipeClicked='true' homePage={hom} recipeData={rec} />);
  //   expect(wrapper.find(Recipe).length).toEqual(1);
  // });

  // it('passes all props to Recipe', () => {
  //   console.log('rec123',rec)
  //   const wrapper = mount(<Home recipeClicked = {true} homePage={hom} recipeData={rec}/>);
  //   const recipeWrapper = wrapper.find(<Recipe homePage={hom} recipeData={rec}/>);
  //   expect(recipeWrapper.find('.recipe')).to.have.lengthOf(1);
  // });
});
