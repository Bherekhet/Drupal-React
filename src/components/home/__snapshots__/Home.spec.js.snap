// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component snapshot renders 1`] = `
<div
  className="main"
>
  <p
    className="intro"
    dangerouslySetInnerHTML={
      Object {
        "__html": "body",
      }
    }
  />
  <div
    className="contain"
  >
    <div
      className="card"
    >
      <div
        className="card-content"
      >
        <span>
          <p
            className="title"
          >
            title section
          </p>
        </span>
        <span>
          <p>
            summary section
          </p>
        </span>
        <a
          href="#"
          onClick={[Function]}
        >
          <img
            alt="title section"
            src="http://gtest.dev.wwbtc.comimage sectio"
          />
        </a>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="card-content"
      >
        <span>
          <p
            className="title"
          >
            title section
          </p>
        </span>
        <span>
          <p>
            summary section
          </p>
        </span>
        <a
          href="#"
          onClick={[Function]}
        >
          <img
            alt="title section"
            src="http://gtest.dev.wwbtc.comimage sectio"
          />
        </a>
      </div>
    </div>
  </div>
</div>
`;
